### PR TITLE
added typescript definitions

### PR DESCRIPTION
**Pull request by Xetera**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    url: string;
+    throwErrors?: boolean;
+    retryOnLimit?: boolean;
+  };
+  export class Webhook {
+    constructor(options: string | WebhookOptions);
+    setUsername(username: string): void;
+    setAvatar(avatar: string): void;
+    sendFile(filePath: string): Promise<void>;
+    send(message: MessageBuilder): Promise<void>;
```
Comment (lleyton) [inline index.d.ts:12]: "message" can also be a string.
```suggestion
    send(message: MessageBuilder | string): Promise<void>;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    info(
+      title: string,
+      fieldName: string,
+      fieldValue: string,
+      inline: boolean
```
Comment (lleyton) [inline index.d.ts:17]: "inline" is optional.
```suggestion
      inline?: boolean
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    ): void;
+    error(
+      title: string,
+      fieldName: string,
+      fieldValue: string,
+      inline?: boolean
+    ): void;
+  }
+  export type WebhookField = {
+    name?: string;
```
Comment (lleyton) [inline index.d.ts:33]: "name" isn't an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-field-structure
```suggestion
    name: string;
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    value?: string;
```
Comment (lleyton) [inline index.d.ts:34]: "value" isn't an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-field-structure
```suggestion
    value: string;
```

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    inline?: boolean;
+  };
+  export type WebhookEmbed = {
+    author?: {
+      name: string;
```
Comment (lleyton) [inline index.d.ts:39]: "name" is an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-author-structure
```suggestion
      name?: string;
```

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      url: string;
```
Comment (lleyton) [inline index.d.ts:40]: "url" is an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-author-structure
```suggestion
      url?: string;
```

File: index.d.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      icon_url: string;
```
Comment (lleyton) [inline index.d.ts:41]: "icon_url" is an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-author-structure
```suggestion
      icon_url?: string;
```

File: index.d.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    };
+    title?: string;
+    url?: string;
+    thumbnail?: {
+      url: string;
```
Comment (lleyton) [inline index.d.ts:46]: "url" is an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-thumbnail-structure
```suggestion
      url?: string;
```

File: index.d.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    };
+    image?: {
+      url: string;
```
Comment (lleyton) [inline index.d.ts:49]: "url" is an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-image-structure
```suggestion
      url: string;
```

File: index.d.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    };
+    timestamp?: Date;
+    color?: number;
+    description?: string;
+    fields: WebhookField[];
+    footer?: {
+      text?: string;
```
Comment (lleyton) [inline index.d.ts:56]: "text" isn't an optional field.
https://discord.com/developers/docs/resources/channel#embed-object-embed-footer-structure
```suggestion
      text: string;
```

File: index.d.ts [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    };
+  };
+
+  export type WebhookPayload = {
+    embeds: WebhookEmbed[];
+  };
+
+  export class MessageBuilder {
+    constructor();
+    getJson(): WebhookPayload;
```
Comment (lleyton) [inline index.d.ts:67]: The function is called getJSON, not getJson.
https://github.com/matthew1232/discord-webhook-node/blob/aa28bcc8d76e46e55d5f8041098608c0572dbe3b/src/classes/messageBuilder.js#L8
```suggestion
    getJSON(): WebhookPayload;
```

File: index.d.ts [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    setText(text: string): this;
+    setAuthor(author: string, authorImage?: string, authorUrl?: string): this;
```
Comment (lleyton) [inline index.d.ts:69]: Name can be optional.
https://discord.com/developers/docs/resources/channel#embed-object-embed-author-structure
```suggestion
    setAuthor(author?: string, authorImage?: string, authorUrl?: string): this;
```

File: index.d.ts [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    setTitle(title: string): this;
+    setUrl(url: string): this;
+    setThumbnail(thumbnailUrl: string): this;
```
Comment (lleyton) [inline index.d.ts:72]: The setImage function is missing.
```suggestion
    setThumbnail(thumbnailUrl: string): this;
    setImage(image: string): this;
```

File: index.d.ts [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+    setTimestamp(): this;
+    setColor(color: number): this;
+    setDescription(description: number): this;
```
Comment (lleyton) [inline index.d.ts:75]: Shouldn't setDescription take a string instead of a number?
```suggestion
    setDescription(description: string): this;
```

**Comment by Xetera** [comment]
Thanks for the feedback, I ended up switching to another library after the PR so that's probably why it ended up being sloppy

File: index.d.ts (hunk 9 above)
Comment (lleyton) [reply index.d.ts:49]: Oops, let me fix my suggestion.
```suggestion
      url?: string;
```

File: index.d.ts (hunk 13 above)
Comment (lleyton) [reply index.d.ts:72]: Reference: https://github.com/matthew1232/discord-webhook-node/blob/aa28bcc8d76e46e55d5f8041098608c0572dbe3b/src/classes/messageBuilder.js#L46

**Comment by lleyton** [comment]
No problem! Can you take a look at [Suggestion 9](https://github.com/matthew1232/discord-webhook-node/pull/12#discussion_r449920428) and [Suggestion 13](https://github.com/matthew1232/discord-webhook-node/pull/12#discussion_r449920645)?

**Comment by lleyton** [comment]
Any reason why you didn't include [Suggestion 13](https://github.com/matthew1232/discord-webhook-node/pull/12#discussion_r449920645)? I might be missing something there.

**Comment by Xetera** [comment]
it's already committed lol https://github.com/matthew1232/discord-webhook-node/pull/12/files#diff-b52768974e6bc0faccb7d4b75b162c99R73

**Comment by lleyton** [comment]
Sorry, my mistake.

**Pull request merged**